examples/lstm_stateful.py

```diff
@@ -0,0 +1,204 @@
+'''Example script showing how to use stateful RNNs
+to model long sequences efficiently.
+Depending on the parameters, either only the stateful LSTM converges,
+or both the stateful and stateless LSTM converge.
```

Comment:
Please flesh out the docstring by explaining what the example is about: what the task is, what the data look like, how the model is set up. Users should be able to go into the code already understanding what they're going to be looking at.